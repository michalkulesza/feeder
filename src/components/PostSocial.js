import React from "react";
import "./PostSocial.scss";
import { Link } from "react-router-dom";

const PostSocial = ({
  likes,
  likesUid,
  likesUsers,
  currentUid,
  handleLike,
  handleDislike,
  username,
  author,
  postKey,
  disableLike
}) => {
  return (
    <div className="post-social">
      <div className="post-social-liked">
        {likesUsers && likesUsers.length > 0 && <span>Liked by: </span>}
        {likesUsers && likesUsers.length > 0
          ? likesUsers.map(user => {
              return (
                <Link to={`/user/${user}`} key={new Date().getTime() * Math.random()}>
                  <div id="social-liked-by">{user}</div>
                </Link>
              );
            })
          : ""}
      </div>
      <span>
        <p>Likes: {likes}</p>
        {disableLike === true ? null : likes > 0 && likesUid.length > 0 ? (
          likesUid.map(like => {
            if (like === currentUid) {
              return (
                <div
                  className="button red"
                  onClick={() => handleDislike(username, currentUid, postKey)}
                  key={new Date().getTime() * Math.random()}
                >
                  Dislike
                </div>
              );
            } else {
              return (
                <div
                  className="button"
                  onClick={() => handleLike(username, currentUid, postKey)}
                  key={new Date().getTime() * Math.random()}
                >
                  Likex
                </div>
              );
            }
          })
        ) : (
          <div
            className="button"
            onClick={() => handleLike(username, currentUid, postKey)}
            key={new Date().getTime() * Math.random()}
          >
            Like
          </div>
        )}
      </span>
    </div>
  );
};

export default PostSocial;
