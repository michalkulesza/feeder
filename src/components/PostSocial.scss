@import "../main.scss";

.post-social {
  height: 30px;
  max-width: 100%;
  margin: 0 auto;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  margin-top: 12px;
  transition: all 0.2s ease-in-out;
  font-size: 0.8rem;
  color: $gray;

  .button {
    margin: 0 10px;
  }

  span {
    max-width: 200px;
    display: flex;
    justify-content: space-between;
    align-items: center;
    cursor: default;
    user-select: none;

    p {
      width: 60px;
    }
  }

  .post-social-liked {
    width: 100%;
    height: 100%;
    display: flex;
    align-items: center;

    span {
      margin-right: 5px;
      cursor: default;
      user-select: none;
    }
  }
}

#social-liked-by {
  cursor: pointer;
  padding: 2px 5px;
  border-radius: 2px;

  &:hover {
    color: $black;
    background-color: $lightestGray;
  }
}
