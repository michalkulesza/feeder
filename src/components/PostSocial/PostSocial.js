import React from "react";
import uuid from "react-uuid";
import { Link } from "react-router-dom";

import "./PostSocial.scss";

const PostSocial = ({ currentUid, handleLike, handleDislike, username, post }) => {
	return post ? (
		<div className="post-social">
			<div className="post-social-liked">
				{post?.likesUsers && post.likesUsers.length > 0 && <span>Liked by: </span>}
				{post?.likesUsers && post.likesUsers.length > 0
					? post?.likesUsers.map(user => {
							return (
								<Link to={`/user/${user}`} key={uuid()}>
									<div id="social-liked-by">{user}</div>
								</Link>
							);
					  })
					: ""}
			</div>
			<span>
				<p>Likes: {post?.likes}</p>
				{post?.disableLike !== true &&
				post?.likes > 0 &&
				post?.likesUid?.length > 0 &&
				post?.likesUid?.includes(currentUid) ? (
					<div className="button red" onClick={() => handleDislike(username, currentUid, post.id)} key={uuid()}>
						Dislike
					</div>
				) : post.disableLike !== true ? (
					<div className="button" onClick={() => handleLike(username, currentUid, post.id)} key={uuid()}>
						Like
					</div>
				) : null}
			</span>
		</div>
	) : null;
};

export default PostSocial;
