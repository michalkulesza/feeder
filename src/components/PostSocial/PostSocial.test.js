import React from "react";
import { shallow } from "enzyme";
import { Link } from "react-router-dom";

import PostSocial from "./PostSocial";

describe("<PostSocial/>", () => {
	const post = { likesUsers: ["test", "test", "test", "test"], disableLike: false };

	it("should match the snapshot without props", () => {
		const wrapper = shallow(<PostSocial />);

		expect(wrapper.html()).toMatchSnapshot();
	});

	it("should render likes", () => {
		const wrapper = shallow(<PostSocial post={post} />);

		expect(wrapper.find(Link).length).toBe(4);
	});

	it("should call handleLike", () => {
		const handleLike = jest.fn();

		const wrapper = shallow(<PostSocial post={post} handleLike={handleLike} />);
		const button = wrapper.find(".button");

		button.simulate("click");

		expect(handleLike).toBeCalled();
	});
});
