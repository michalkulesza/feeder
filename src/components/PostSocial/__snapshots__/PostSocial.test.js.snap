// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostSocial/> should match the snapshot without props 1`] = `null`;
